docs/designers-developers/developers/block-api/block-supports.md

```diff
@@ -194,6 +194,45 @@ supports: {
 }
 ```
 
+## duotone
+
+- Type: `boolean`, `string`, `string[]`, or `object`
+- Default value: `false`
+
+This property adds block controls which allow to apply a duotone filter to a block or part of a block. _Important: It doesn't work with dynamic blocks yet._
+
+```js
+supports: {
+    // Declare support for duotone.
+    // This applies the duotone filter to the entire block in both edit and save.
+    duotone: true
+}
+```
+
+```js
+supports: {
+    // Apply the filter to the same selector in both edit and save.
+    duotone: 'img'
+}
+```
+
+```js
+supports: {
+    // Apply the filter to both images and videos
+    duotone: ['img', 'video']
+}
+```
+
+```js
+supports: {
+    // Apply the filter to a specific inner element in edit and the whole block in save.
+    duotone: {
+        edit: '.duotone-image img'
+        save: true
```

Comment:
I'm not sure I like this API personally, our goal is always to have "edit" match "save" which means it's better if we try to align the markup instead of offering APIs to encourage different ones.

Reply:
We're already kind of in this situation with having some styles that only apply in the editor. And I want to make sure that, if we have an editing mode—like the image cropper—that requires extra markup strictly in the editing stage, we can still apply the styles in that mode. That additional UI may come from a third-party component where you can't add a classname to match the frontend.

Reply:
> That additional UI may come from a third-party component where you can't add a classname to match the frontend.

I think I have the same issue in the group block in #29335 
That said, I do believe the API should still stay pure. We have a few options I think:

 - Try to refactor that extra bit of UI to don't impact the markup: Only adds markup but don't break the block structure or alternatively use a "Popover" for it similar to how we do with the block toolbar where it's shown by the block but in reality it's not there in the DOM. If we find that it's a common use-case we might to provide a simple way to do that for any UI/component.
 - Provide a flag to not apply the support flag automatically: have the support flag add the UI to the toolbar, saving the attribute... but applying the attribute is left to the block author. @nosolosw is also working on a similar API for other block supports.

Reply:
The API Riad refers to has already landed for `color`. See the PRs listed under "skip serialization" at https://github.com/WordPress/gutenberg/issues/28913

Reply:
@nosolosw @youknowriad I need to use the child combinator (`>`) on the parent block for [cover block nesting](https://github.com/WordPress/gutenberg/pull/26752/commits/64b12777ecd0853ad712d9aef92918248561d925), should we support using the Sass/SCSS parent selector (`&`) for that? Right now I have it added automatically to prevent inner blocks from getting the duotone filter applied, but nesting a cover block inside a cover block breaks it. Really, the parent selector should always be used, but sometimes it just needs to be applied with a different combinator.